app/src/main/java/org/mozilla/focus/utils/Browsers.java

```diff
@@ -237,12 +240,31 @@ public boolean isDefaultBrowser(Context context) {
 
     }
 
-    public ActivityInfo[] getInstalledBrowsers() {
-        final Collection<ActivityInfo> collection = browsers.values();
+    public ActivityInfo[] getInstalledBrowsers(boolean isCustomTab) {
```

Comment:
I think you chose the optimal implementation strategy, given the existing code, but I just want to make a note: `Browsers` probably shouldn't be an object that needs to be constructed and these should all be static methods. The current implementation creates some strangeness:
-  Some values are cached and some aren't
- Holding a long-lived reference to the object can make the cached values out of date.
- We have to [explicitly remove Focus from the initial cached browsers value](https://github.com/mozilla-mobile/focus-android/commit/c2664c9b12ad9e18907ada8c7af84620943b57fb) and then re-add it here in this method.

I assume the reason we do it this way is to allow mocking but I'd rather use PowerMockito to mock static methods than mush the code up like this. :)

Reply:
Note that the data in this object is based on the URL. The set of default apps/browsers can be different depending on what URL you want to open. That's the major reason why some of this data is not cached statically.

Reply:
To be explicit, I was suggesting we should never cache this data (because we do in a Browsers instance) and use static functions instead.

Reply:
Generating this data is (or can be) super slow though. :)